File: web/src/plugin/translist/api/job.js
import service from '@/utils/request'

export const uploadAndTranslate = (data) => {
  return service({
    url: '/translist/uploadAndTranslate',
    method: 'post',
    data,
    headers: { 'Content-Type': 'multipart/form-data' },
    donNotShowLoading: true
  })
}

export const deleteJob = (params) => {
  return service({
    url: '/translist/deleteJob',
    method: 'delete',
    params
  })
}

export const deleteJobByIds = (params) => {
  return service({
    url: '/translist/deleteJobByIds',
    method: 'delete',
    params
  })
}

export const findJob = (params) => {
  return service({
    url: '/translist/findJob',
    method: 'get',
    params
  })
}

export const getJobList = (params) => {
  return service({
    url: '/translist/getJobList',
    method: 'get',
    params
  })
}

export const retranslate = (params) => {
  return service({
    url: '/translist/retranslate',
    method: 'post',
    params
  })
}

export const exportExcel = (params) => {
  return service({
    url: '/translist/exportExcel',
    method: 'get',
    params,
    responseType: 'blob'
  })
}

File: server/plugin/translist/router/job.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Job = new(job)

type job struct{}

func (r *job) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("translist").Use(middleware.OperationRecord())
		group.POST("uploadAndTranslate", apiJob.UploadAndTranslate)
		group.DELETE("deleteJob", apiJob.DeleteJob)
		group.DELETE("deleteJobByIds", apiJob.DeleteJobByIds)
		group.POST("retranslate", apiJob.Retranslate)
	}
	{
		group := private.Group("translist")
		group.GET("findJob", apiJob.FindJob)
		group.GET("getJobList", apiJob.GetJobList)
		group.GET("exportExcel", apiJob.ExportExcel)
	}
	_ = public
}
